python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -834,10 +942,15 @@ def _log_iteration_timestamp(
     # Rather than logging every single iteration, we log every hour.  This means that if the cron
     # schedule changes to run to a time that is less than an hour ago, when the code location is
     # deployed, a tick might be registered for that time, with a run kicking off.
+
+    # Returns the next timestamp that we should plan to log the last_iteration_timestamp - with some
+    # additional jitter so that threads won't all come back at the exact same time
+    random_jitter_offset = random.randint(0, LAST_RECORDED_ITERATION_JITTER_SECONDS)
```

Comment:
I don't have hard numbers from before and after, but I have data that indicates that for large workspaces the first tick is much slower since it needs to spin up a thread for every schedule. Without this jitter it seems that you would pay that cost again every LAST_RECORDED_ITERATION_INTERVAL_SECONDS seconds.

Reply:
I don't think they are spread out? The way i understand it is that on startup, we iterate over all, say, 5k schedules with a fixed 'now' timestamp. For most of them it will not be time to run, and we will write out a fixed offset for 'now' (fixed for everybody) + LAST_RECORDED_ITERATION_INTERVAL_SECONDS. So by default you end up thundering every LAST_RECORDED_ITERATION_INTERVAL_SECONDS seconds if you're not careful. Agree it makes things more complex and will update comment